### DISCUSSION TITLE
c# : Use Two foreach loop in Single Table add to record:  Record column is misalign.

### DISCUSSION THREAD

**Discussion by pradeepInnovex** [QuestPDF/QuestPDF#826] [Q&A] [answered]
This is my Table : When I use two foreach loop, table cell is not align properly : This is my Code : 
     .Table(table =>
     {
         table.ColumnsDefinition(columns =>
         {

             columns.ConstantColumn(6, Unit.Millimetre);
             columns.ConstantColumn(10, Unit.Millimetre);
             columns.ConstantColumn(6, Unit.Millimetre);
             columns.ConstantColumn(10, Unit.Millimetre);
             columns.ConstantColumn(10, Unit.Millimetre);
           //  columns.ConstantColumn(30, Unit.Millimetre);
             columns.ConstantColumn(6, Unit.Millimetre);
             columns.ConstantColumn(10, Unit.Millimetre);
             columns.ConstantColumn(12, Unit.Millimetre);
             columns.ConstantColumn(15, Unit.Millimetre);
             columns.ConstantColumn(15, Unit.Millimetre);
             columns.ConstantColumn(15, Unit.Millimetre);
             columns.ConstantColumn(15, Unit.Millimetre);
             columns.ConstantColumn(15, Unit.Millimetre);
             columns.ConstantColumn(15, Unit.Millimetre);
             columns.ConstantColumn(15, Unit.Millimetre);

         });

         table.Header(header =>
         {
             header.Cell().ColumnSpan(2).Border(1, Unit.Point).AlignCenter().Text("In").FontSize(8);
             header.Cell().ColumnSpan(2).Border(1, Unit.Point).AlignCenter().Text("Out").FontSize(8);
             header.Cell().Border(1, Unit.Point).Padding(1, Unit.Point).AlignCenter().Text("Shift").FontSize(8);
            // header.Cell().Border(1, Unit.Point).Padding(3, Unit.Point).AlignCenter().Text("Activity").FontSize(8);
             header.Cell().ColumnSpan(2).Border(1, Unit.Point).AlignCenter().Text("Operator | Helper").FontSize(8);
             header.Cell().Border(1, Unit.Point).AlignCenter().Text("Qty / Emp").FontSize(8);
             header.Cell().Border(1, Unit.Point).AlignCenter().Text("Rate").FontSize(8);
             header.Cell().Border(1, Unit.Point).AlignCenter().Text("Daily Pay").FontSize(8);
             header.Cell().Border(1, Unit.Point).AlignCenter().Text("Ot").FontSize(8);
             header.Cell().Border(1, Unit.Point).AlignCenter().Text("Daily Pay + Ot").FontSize(8);
             header.Cell().Border(1, Unit.Point).AlignCenter().Text("Piece Rate").FontSize(8);
             header.Cell().Border(1, Unit.Point).AlignCenter().Text("Old Rate").FontSize(8);
             header.Cell().Border(1, Unit.Point).AlignCenter().Text("Test").FontSize(8);

             
         });

         foreach (var item1 in item.DayDetaiPieceRates) // item1.InDate != DateTime.MinValue ? item1.InDate.Day.ToString() : ""
         {
             table.Cell().EnsureSpace().Border(1, Unit.Point).Padding(3, Unit.Point).AlignCenter().Text(item1.InDate != DateTime.MinValue ? item1.InDate.Day.ToString():"").FontSize(8);
             table.Cell().EnsureSpace().Border(1, Unit.Point).Padding(3, Unit.Point).AlignCenter().Text($"{item1.InTime.ToString(@"hh\:mm")}").FontSize(8).FontFamily("Roboto");
             table.Cell().EnsureSpace().Border(1, Unit.Point).Padding(3, Unit.Point).AlignCenter().Text(item1.InDate != DateTime.MinValue ? item1.OutDate.Day.ToString() : "").FontSize(8).FontFamily("Roboto");
             table.Cell().EnsureSpace().Border(1, Unit.Point).Padding(3, Unit.Point).AlignCenter().Text($"{item1.OutTime.ToString(@"hh\:mm")}").FontSize(8).FontFamily("Roboto");
             table.Cell().EnsureSpace().Border(1, Unit.Point).Padding(3, Unit.Point).AlignCenter().Text($"{item1.ShiftName}").FontSize(6).FontFamily("Roboto");
           //  table.Cell().EnsureSpace().Border(1, Unit.Point).Padding(3, Unit.Point).AlignCenter().Text($"{item1.ActivityName} - {item1.Qty:F2}").FontSize(8).FontFamily("Roboto");
             table.Cell().EnsureSpace().Border(1, Unit.Point).Padding(3, Unit.Point).AlignCenter().Text(item1.NumberOfOperators).FontSize(8).FontFamily("Roboto");
             table.Cell().EnsureSpace().Border(1, Unit.Point).Padding(3, Unit.Point).AlignCenter().Text(item1.NumberOfHelpers).FontSize(8).FontFamily("Roboto");
             table.Cell().EnsureSpace().Border(1, Unit.Point).Padding(3, Unit.Point).AlignCenter().Text($"{item1.QuantityPerEmployee:F2}").FontSize(8).FontFamily("Roboto");
             table.Cell().EnsureSpace().Border(1, Unit.Point).Padding(3, Unit.Point).AlignCenter().Text($"{item1.Rate:F2}").FontSize(8).FontFamily("Roboto");
             table.Cell().EnsureSpace().Border(1, Unit.Point).Padding(3, Unit.Point).AlignCenter().Text($"{item1.DailySalary:F2}").FontSize(8).FontFamily("Roboto");
             table.Cell().EnsureSpace().Border(1, Unit.Point).Padding(3, Unit.Point).AlignCenter().Text($"{item1.OverTimeAmount:F2}").FontSize(8).FontFamily("Roboto");
             table.Cell().EnsureSpace().Border(1, Unit.Point).Padding(3, Unit.Point).AlignCenter().Text($"{item1.DailySalryWith_Ot:F2}").FontSize(8).FontFamily("Roboto");
             table.Cell().EnsureSpace().Border(1, Unit.Point).Padding(3, Unit.Point).AlignCenter().Text($"{item1.PayingSalary:F2}").FontSize(8).FontFamily("Roboto");
             table.Cell().EnsureSpace().Border(1, Unit.Point).Padding(3, Unit.Point).AlignCenter().Text($"{item1.OldRateSalary:F2}").FontSize(8).FontFamily("Roboto");
             table.Cell().EnsureSpace().Border(1, Unit.Point).Padding(3, Unit.Point).AlignCenter().Text($"{item1.OldRateSalary:F2}").FontSize(8).FontFamily("Roboto");

         }

         foreach (var item in item.DailySums)
         {
             table.Cell().EnsureSpace().Border(1, Unit.Point).Padding(3, Unit.Point).AlignCenter().Text($"{item1.OldRateSalary:F2}").FontSize(8).FontFamily("Roboto");
         }

         
     });

**Comment by DawidRobertNowak** [comment, marked as answer]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Regarding the misalignment — QuestPDF tables fill cells left-to-right based on the column count. The first `foreach` loop adds 16 cells per iteration (matching the 16 defined columns), so rows align correctly. However, the second `foreach` (`DailySums`) adds only 1 cell per iteration, causing subsequent cells to fill into unexpected columns. To fix this, each iteration should emit a cell for every column, or use `ColumnSpan()` to span the desired width.

You can find more details and examples in the table documentation:
https://www.questpdf.com/api-reference/table/basics.html

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
